### PR TITLE
Added dict key validation & updated documentation

### PR DESCRIPTION
**Pull request by kanekosh**
### Key validation
With this PR, OAS will check a user-defined dicts (mesh dict or surface dict) and warn if the user provides a key that is not implemented or used in OAS.

For example, some examples and tests provide the `chord` key when creating a mesh, but the key `chord` does not exist and therefore never used. I fixed them to `root_chord`, the one actually used.
I also removed unused keys from other tests.

### Documentation update
The current documentation and docstrings say the `chord_cp` variable is the actual chord value in meters, but what the code (ScaleX component) actually does is that it applies the chord scaler to the initial mesh. Therefore `chord_cp` is actually a dimensionless chord scaler.  
This PR fixes the documentation and docstrings.

Also, I updated the surface dict documentation.

(It's a little too late but I probably should've opened two separate PRs for this...)

## Type of change

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing

I added a new test to check the chord values before and after `ScaleX` is applied.

## Checklist

- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [x] I have run unit and regression tests which pass locally with my changes
- [x] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by eytanadler — DISMISSED** [review]
Thanks for fixing these!

**Comment by nwu63** [comment]
> In addition, I found that some examples and tests provide the chord key when creating a mesh, but the key chord does not exist and therefore never used. I fixed them to root_chord, the one actually used.

Should there be some sort of key validation then? To ensure we aren't passing in dict entries that are silently inactive?

**Comment by kanekosh** [comment]
> > In addition, I found that some examples and tests provide the chord key when creating a mesh, but the key chord does not exist and therefore never used. I fixed them to root_chord, the one actually used.
> 
> Should there be some sort of key validation then? To ensure we aren't passing in dict entries that are silently inactive?

That's true, I'll add it.

**Comment by kanekosh** [comment]
Added key validation as @nwu63 suggested, ready for the 2nd review

File: openaerostruct/utils/check_surface_dict.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    Key valication function for the OAS surface dict.
+    Shows a warning if a user provided a key that is (likely) not implemented in OAS.
+
+    Parameters
+    ----------
+    surface : dict
+        User-defined surface dict
+    """
+
+    # NOTE: make sure this is consistend to the documentation's surface dict page
```
Comment (eytanadler) [inline openaerostruct/utils/check_surface_dict.py:15]: Consistend -> consistent

**Review by eytanadler — DISMISSED** [review]
Thanks for doing this tedious work. Only one very minor spelling correction, and otherwise looks good!

**Review by eytanadler — APPROVED** [review]

**Comment by kanekosh** [comment]
Thanks for the quick review!

**Pull request merged**